pkg/controller/shoot/shoot_quota_control.go

```diff
@@ -0,0 +1,195 @@
+package shoot
+
+import (
+	"fmt"
+	"time"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	gardeninformers "github.com/gardener/gardener/pkg/client/garden/informers/externalversions/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (c *Controller) shootQuotaAdd(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return
+	}
+	c.shootQuotaQueue.Add(key)
+}
+
+func (c *Controller) shootQuotaDelete(obj interface{}) {
+	shoot, ok := obj.(*gardenv1beta1.Shoot)
+	if shoot == nil || !ok {
+		return
+	}
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return
+	}
+	c.shootQuotaQueue.Done(key)
+}
+
+func (c *Controller) reconcileShootQuotaKey(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	shoot, err := c.shootLister.Shoots(namespace).Get(name)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
```

Comment:
Any reason why we don't have log messages here like in the [other controllers](https://github.com/gardener/gardener/blob/master/pkg/controller/shoot/shoot_maintenance_control.go#L62-L66)?

Reply:
done.